src/main/kotlin/com/pubnub/api/presence/Presence.kt

```diff
@@ -33,7 +33,7 @@ internal interface Presence {
             executorService: ScheduledExecutorService
         ): Presence {
             if (heartbeatInterval <= Duration.ZERO || !enableEventEngine) {
-                return PresenceNoOp()
+                return PresenceNoOp(if (!suppressLeaveEvents) leaveProvider else null)
```

Comment:
I am thinking to pass to PresenceNoOp both suppressLeaveEvents and leaveProvider and inside PresenceNoOp have logic that is responsible for sending leave event. 
That way PresenceNoOp is responsible for sending leave event if needed instead of external code.
What do you think?

Reply:
done